scripts/coverage_ratchet.py

```
@@ -48,18 +38,34 @@ def _percent(metric: Metric) -> float:
+
+
+def _compare_policy(report_name: str, candidate: Any, reference: Any) -> list[str]:
+    if not isinstance(candidate, dict) or not isinstance(reference, dict):
+        return [f"{report_name}: baseline measurement policy is malformed"]
+    errors: list[str] = []
+    for field in ("all", "all_targets", "branch", "include_namespace_packages", "workspace"):
+        if reference.get(field) is True and candidate.get(field) is not True:
+            errors.append(f"{report_name}: measurement policy disabled {field}")
+    for field in ("features", "source_dirs", "source_pkgs", "tests"):
+        if not set(reference.get(field, [])).issubset(set(candidate.get(field, []))):
+            errors.append(f"{report_name}: measurement policy narrowed {field}")
```

Ruff (0.16.3)

[warning] Use `list.extend` to create a transformed list

(PERF401)